### PR TITLE
python312Packages.radio-beam: Fix build

### LINKED ISSUE #298597: Build failure: python312Packages.radio-beam
### Steps To Reproduce

1. `nix-build -A python312Packages.radio-beam`

### Build log

```console
Running phase: pytestCheckPhase
Executing pytestCheckPhase
ImportError while loading conftest '/private/tmp/nix-build-python3.12-radio-beam-0.3.7.drv-0/radio-beam-0.3.7/radio_beam/conftest.py'.
radio_beam/conftest.py:7: in <module>
    from distutils.version import LooseVersion
E   ModuleNotFoundError: No module named 'distutils'
```

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.4.0, macOS 14.4`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.21.0`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by smaret**
## Description of changes

Closes #298597.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by smaret** [comment]
Result of `nixpkgs-review pr 298610` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>python311Packages.glueviz</li>
    <li>python311Packages.glueviz.dist</li>
    <li>python311Packages.pvextractor</li>
    <li>python311Packages.pvextractor.dist</li>
    <li>python311Packages.spectral-cube</li>
    <li>python311Packages.spectral-cube.dist</li>
    <li>python312Packages.spectral-cube</li>
    <li>python312Packages.spectral-cube.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.radio-beam</li>
    <li>python311Packages.radio-beam.dist</li>
    <li>python312Packages.radio-beam</li>
    <li>python312Packages.radio-beam.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/radio-beam/default.nix [hunk 1]
```diff
@@ -21,6 +21,10 @@ buildPythonPackage rec {
     hash = "sha256-7AFkuuYLzibwwgz6zrFw0fBXCnGLzdm4OgT+Chve5jU=";
   };
 
+  # Fix distutils deprecation in Python 3.12. See:
+  # https://github.com/radio-astro-tools/radio-beam/pull/124
+  patches = [ ./fix-distutil-depr.patch ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/radio-beam/default.nix:26]: Why not use `fetchpatch2`?

Comment (smaret) [reply pkgs/development/python-modules/radio-beam/default.nix:26]: Thanks for the suggestion. But I thought the policy was to fetch patches from the upstream repository only?  The PR on upstream isn't accepted yet.

Comment (smaret) [reply pkgs/development/python-modules/radio-beam/default.nix]: @dotlambda The PR has been merged upstream.

**Comment by smaret** [comment]
Result of `nixpkgs-review pr 298610` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>python311Packages.glueviz</li>
    <li>python311Packages.glueviz.dist</li>
    <li>python311Packages.pvextractor</li>
    <li>python311Packages.pvextractor.dist</li>
    <li>python311Packages.spectral-cube</li>
    <li>python311Packages.spectral-cube.dist</li>
    <li>python312Packages.spectral-cube</li>
    <li>python312Packages.spectral-cube.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.radio-beam</li>
    <li>python311Packages.radio-beam.dist</li>
    <li>python312Packages.radio-beam</li>
    <li>python312Packages.radio-beam.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/radio-beam/default.nix [hunk 2]
```diff
@@ -21,6 +22,15 @@ buildPythonPackage rec {
     hash = "sha256-7AFkuuYLzibwwgz6zrFw0fBXCnGLzdm4OgT+Chve5jU=";
   };
 
+  # Fix distutils deprecation in Python 3.12. See:
+  # https://github.com/radio-astro-tools/radio-beam/pull/124
+  patches = [
+    (fetchpatch2 {
+      url = "https://github.com/radio-astro-tools/radio-beam/commit/1eb0216c8d7f5a4494d8d1fe8c79b48425a9c491.patch";
+      sha256 = "sha256-tAuC4EVuSodWy9JvvfBIVhbqM/wFkqikq2Q2rFczeRI=";
```
Comment (dotlambda) [inline pkgs/development/python-modules/radio-beam/default.nix:30]:
```suggestion
      hash = "sha256-tAuC4EVuSodWy9JvvfBIVhbqM/wFkqikq2Q2rFczeRI=";
```

**Review by dotlambda — COMMENTED** [review]
Please squash the commits. Btw the package is called radio-beam, not radio_beam.

**Comment by wegank** [comment]
@ofborg build python312Packages.radio-beam python312Packages.radio-beam.passthru.tests

**Pull request merged**